src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3019,8 +3022,26 @@ private BoundExpression BindArgumentExpression(BindingDiagnosticBag diagnostics,
                     }
                 }
             }
+
+            void reportUnsafeIfNeeded(MemberResolutionResult<TMember> methodResult, BindingDiagnosticBag diagnostics, BoundExpression argument, TypeWithAnnotations parameterTypeWithAnnotations)
+            {
+                // NOTE: for some reason, dev10 doesn't report this for indexer accesses.
```

Comment:
Do we need to maintain such legacy behavior for new scenario? I understand that `if (!kind.IsIdentity)` case needs to stay like this, but the `if (kind.IsInterpolatedStringHandler)` is not bound by prior behavior.

Reply:
We could potentially do this, but I'd rather just keep it simple and have one set of unsafe binding rules for everything.